samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -7,7 +7,12 @@
 import re
 import logging
 
-from samcli.hook_packages.terraform.hooks.prepare.exceptions import InvalidResourceLinkingException
+from samcli.hook_packages.terraform.hooks.prepare.exceptions import (
+    InvalidResourceLinkingException,
+    OneLambdaLayerLinkingLimitationException,
+    LocalVariablesLinkingLimitationException,
+)
+from samcli.hook_packages.terraform.hooks.prepare.hook import _build_cfn_logical_id
```

Comment:
This might cause as circular import when we import a function from this package to hooks. We should move `_build_cfn_logical_id` to a third package.

Reply:
Yes, I was thinking to do this change when we call this function. Any way, I will do it in this PR.

Reply:
done